File: spring/mapstruct-spring-extensions-example/src/main/java/com/github/yukihane/spring/mapstructspringextensionsexample/PedalMapper.java
package com.github.yukihane.spring.mapstructspringextensionsexample;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.springframework.core.convert.converter.Converter;

@Mapper
public interface PedalMapper extends Converter<Pedal, PedalDto> {
    @Override
    @Mapping(source = "size", target = "pedalSize")
    PedalDto convert(Pedal pedal);
}

File: spring/mapstruct-spring-extensions-kotlin/src/main/kotlin/com/example/mapstruct_spring_extensions_kotlin/PedalMapper.kt
package com.example.mapstruct_spring_extensions_kotlin

import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.springframework.core.convert.converter.Converter

@Mapper
interface PedalMapper : Converter<Pedal, PedalDto> {
    @Mapping(source = "size", target = "pedalSize")
    override fun convert(pedal: Pedal): PedalDto
}
